src/uu/df/src/table.rs

```diff
@@ -102,8 +102,8 @@ impl AddAssign for Row {
         let bytes = self.bytes + rhs.bytes;
         let bytes_used = self.bytes_used + rhs.bytes_used;
         let bytes_avail = self.bytes_avail + rhs.bytes_avail;
-        let inodes = self.inodes + rhs.inodes;
-        let inodes_used = self.inodes_used + rhs.inodes_used;
+        let inodes = self.inodes.saturating_add(rhs.inodes);
+        let inodes_used = self.inodes_used.saturating_add(rhs.inodes_used);
```

Comment:
Do we really have sometimes a number of inodes greater than `u64::MAX`?

If this is the case, is it ok to compute a *wrong* value by saturating the addition?

Reply:
this peace of code does an accumulation. As far as I understood the code, this accumulation is purely needed to compute an optional finalising row called "total" in the output. This "total" row shall display the sum of each column. Thus, also the inodes are accumulated.

On my phones system, there are apparently several filesystems that reports "inodes" as exactly `u64::MAX`. Summing that up always causes an int 64 overflow. But is it relevant? I guess not really as the number is anyway insanely high. If someone finds a usecase where this is relevant, we can still extend the integer type to `u128`.
Alternatively. one could automatically filter the filesystem with `u64::MAX` because the number might anyway not be valid.

Output of GNU version of df on my phone:
```
~/rust/coreutils $ df -i
Filesystem                                           Inodes  IUsed                IFree IUse% Mounted on
/dev/block/dm-15                       18446744073709551615   3141 18446744073709548474    1% /
tmpfs                                               1457181   1464              1455717    1% /dev
tmpfs                                               1457181     65              1457116    1% /mnt
/dev/block/dm-16                       18446744073709551615   3002 18446744073709548613    1% /system_ext
/dev/block/dm-17                                        304    298                    6   99% /product
overlay-overlay                        18446744073709551615      0 18446744073709551615    0% /product/app
overlay-overlay                        18446744073709551615      0 18446744073709551615    0% /product/priv-app
overlay-overlay                        18446744073709551615      0 18446744073709551615    0% /product/lib64
overlay-overlay                        18446744073709551615      0 18446744073709551615    0% /product/lib
overlay-overlay                        18446744073709551615      0 18446744073709551615    0% /product/overlay
overlay                                18446744073709551615      0 18446744073709551615    0% /product/etc/permissions
overlay                                18446744073709551615      0 18446744073709551615    0% /product/framework
/dev/block/dm-18                       18446744073709551615      0 18446744073709551615    0% /vendor
/dev/block/dm-19                       18446744073709551615      0 18446744073709551615    0% /odm
/dev/block/dm-20                       18446744073709551615      0 18446744073709551615    0% /my_product
/dev/block/dm-21                       18446744073709551615      0 18446744073709551615    0% /my_engineering
/dev/block/dm-23                       18446744073709551615      0 18446744073709551615    0% /my_carrier
/dev/block/dm-24                       18446744073709551615      0 18446744073709551615    0% /my_region
/dev/block/dm-25                       18446744073709551615      0 18446744073709551615    0% /my_heytap
/dev/block/dm-26                       18446744073709551615      0 18446744073709551615    0% /my_stock
/dev/block/dm-27                       18446744073709551615      0 18446744073709551615    0% /my_preload
/dev/block/dm-28                       18446744073709551615      0 18446744073709551615    0% /my_bigball
/dev/block/dm-29                       18446744073709551615      0 18446744073709551615    0% /my_manifest
tmpfs                                               1457181     99              1457082    1% /apex
/dev/block/loop7                                         48     35                   13   73% /apex/com.android.i18n@1
/dev/block/loop8                                         32     29                    3   91% /apex/com.android.wifi@319999900
/dev/block/loop5                                         32     20                   12   63% /apex/com.android.apex.cts.shim@1
/dev/block/loop6                                         32     23                    9   72% /apex/com.android.appsearch@300000000
/dev/block/loop11                                       304    290                   14   96% /apex/com.android.vndk.v31@1
/dev/block/dm-30                                         32     25                    7   79% /apex/com.android.extservices@341131030
/dev/block/dm-32                                         32     28                    4   88% /apex/com.android.permission@341110010
/dev/block/loop15                                        64     50                   14   79% /apex/com.android.runtime@1
/dev/block/dm-33                                        176    167                    9   95% /apex/com.android.conscrypt@341110002
/dev/block/dm-31                                        112    112                    0  100% /apex/com.android.art@341110062
/dev/block/loop16                                       608    596                   12   99% /apex/com.android.vndk.v30@1
/dev/block/dm-34                                         32     20                   12   63% /apex/com.android.tzdata@313110000
/dev/block/dm-35                                         32     20                   12   63% /apex/com.android.scheduling@340819220
/dev/block/dm-36                                         32     31                    1   97% /apex/com.android.os.statsd@341114000
/dev/block/dm-37                                         64     58                    6   91% /apex/com.android.media@341111002
/dev/block/dm-38                                         32     27                    5   85% /apex/com.android.mediaprovider@341111020
/dev/block/dm-39                                         48     36                   12   75% /apex/com.android.adbd@340912202
/dev/block/dm-40                                        128    114                   14   90% /apex/com.android.media.swcodec@341111002
/dev/block/dm-42                                         32     21                   11   66% /apex/com.android.cellbroadcast@341110000
/dev/block/dm-41                                         16     16                    0  100% /apex/com.android.neuralnetworks@341010000
/dev/block/loop27                                        32     26                    6   82% /apex/com.google.mainline.primary.libs@331059000
/dev/block/dm-43                                         32     20                   12   63% /apex/com.android.ipsec@340914200
/dev/block/dm-44                                         80     66                   14   83% /apex/com.android.tethering@341112070
/dev/block/dm-45                                         80     75                    5   94% /apex/com.google.mainline.primary.libs@341177000
/dev/block/dm-46                                         16     16                    0  100% /apex/com.android.resolv@341110000
/dev/block/dm-47                                         32     24                    8   75% /apex/com.android.sdkext@341110000
/dev/block/bootdevice/by-name/cache                  114688     55               114633    1% /cache
/dev/block/bootdevice/by-name/userdata             12812797 348578             12464219    3% /data
/dev/fuse                                          12812797 348578             12464219    3% /storage/emulated
/data/media                                        12812797 348578             12464219    3% /storage/emulated/0/Android/data
~/rust/coreutils $ 
```

Reply:
This looks awfully similar to a -1 cast to `uint64_t`…

Reply:
Could you retry with `--total` as well? That would be interesting to see how GNU coreutils perform.

Reply:
by the way, the inodes for erofs are not really relevant in general as it is a **read only filesystem**